### PR TITLE
Add ENet SAD-Simple model for LLAMAS

### PR DESCRIPTION
**Pull request by cardwing**
@karstenBehrendt Thanks for your comments. Here, there are several points I want to clarify.

First, the purpose of using ParallelCriterion2.lua is to get the value of each loss term instead of the total one (original criterion function did so).

Second, since this code is only used for academic research, using the BSD License should be just ok. 

Third, I have removed these files (cifar10-gen.lua, cifar10.lua, cifar100-gen.lua, cifar100.lua, imagenet-gen.lua and imagenet.lua) from the original repo since they are indeed irrelevant to the lane detection task.  

Finally, as to many other lua files, I have changed the original files to better serve the lane detection task. Hence, it is inconvenient to remove them from the current code while maintaining the functionality.

### CODE REVIEW

File: ENet-SAD-Simple/datasets/transforms.lua [hunk 1]
```diff
@@ -0,0 +1,385 @@
+--
```
Comment (karstenBehrendt) [inline ENet-SAD-Simple/datasets/transforms.lua]: File is a copy of (or at least very similar to) https://github.com/facebook/fb.resnet.torch/blob/master/datasets/transforms.lua

File: ENet-SAD-Simple/datasets/transforms.lua [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+         return image.scale(input, w, h, interpolation)
+      else
+         target = image.crop(target, 0, 333, 1276, 717)
+         return image.scale(input, w, h, interpolation), image.scale(target, w, h, 'simple'), exist
+      end
+   end
+end
+	
+-- Crop to centered rectangle
+function M.CenterCrop(w, h)
```
Comment (karstenBehrendt) [inline ENet-SAD-Simple/datasets/transforms.lua]: This function seems to be different

File: ENet-SAD-Simple/datasets/transforms.lua [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+      if w < h then
+         return image.scale(input, size, h/w * size, interpolation)
+      else
+         return image.scale(input, w/h * size, size, interpolation)
+      end
+   end
+end
+
+-- Added by PanXingang. Scales the width and height for input and target
+function M.ScaleWH(w, h, interpolation)
```
Comment (karstenBehrendt) [inline ENet-SAD-Simple/datasets/transforms.lua]: This function has been adapted to llamas

**Comment by cardwing** [comment]
Thanks for your advice. I will start another pull request.

**Pull request closed without merging**